src/labone/core/session.py

```diff
@@ -675,13 +681,54 @@ async def subscribe(
             raise TypeError(msg) from None
         request = self._session.subscribe_request()
         request.subscription = subscription
-        response = await _send_and_wait_request(request)
+
+        if get_initial_value:
+            response, initial_value = await asyncio.gather(
+                _send_and_wait_request(request),
+                self.get(path),
+            )
+        else:
+            response = await _send_and_wait_request(request)
+            initial_value = None
+
         unwrap(response.result)  # Result(Void, Error)
         new_queue_type = queue_type or DataQueue
-        return new_queue_type(
+        queue = new_queue_type(
             path=path,
             register_function=streaming_handle.register_data_queue,
         )
+        if initial_value is not None:
```

Comment:
nitpick: Can we get around with one `if`-statement caused by `get_initial_value`? Nitpick as it would probably result to a few duplicate lines in the function.